### PR TITLE
Implement getRobotVelocity on RobotInformation and use on executions instead of tf

### PR DESCRIPTION
**Pull request by corot**
How can I run locally run-style-check-diff.sh so I format only the modified lines?

Our system is a bit cumbersome... as I need to format just the modified lines :thinking: 
So I'll let the previous format

### CODE REVIEW

File: mbf_abstract_nav/include/mbf_abstract_nav/abstract_controller_execution.h
Comment (corot) [inline mbf_abstract_nav/include/mbf_abstract_nav/abstract_controller_execution.h]: It should be a const reference, but bloody OdometryHelper doesn't declare its methods as const
Options:
- PR OdometryHelper: will take ages to merge, if ever
- Copy it onto mbf_utility and modify
- Let it this way

If nothing against, I'l go for the 1st and 2nd options, on a separated commit for easy reverting, in case 1st gets merged fast.
Sounds good?  @spuetz , @dorezyuk

Comment (dorezyuk) [reply mbf_abstract_nav/include/mbf_abstract_nav/abstract_controller_execution.h]: A better solution is to declare the odom helper in the RobotInformation as mutable (and thus make the methods using the odom helper const)

Comment (corot) [reply mbf_abstract_nav/include/mbf_abstract_nav/abstract_controller_execution.h]: ah...  good idea!  changed

File: mbf_utility/src/robot_information.cpp
Comment (dorezyuk) [inline mbf_utility/src/robot_information.cpp:79]: would it make sense to check here the data? something like 
```c++
if(!base_odom.header.seq && base_odom.header.stamp == ros::Time(0)){
  ROS_WARN("No odometry message received");
  return false;
}
```

File: mbf_utility/CMakeLists.txt
Comment (dorezyuk) [inline mbf_utility/CMakeLists.txt]: Didn't we had some issues with this dependency? Or why did we copy the footprint helper?

File: mbf_abstract_nav/include/mbf_abstract_nav/abstract_controller_execution.h
Comment (dorezyuk) [inline mbf_abstract_nav/include/mbf_abstract_nav/abstract_controller_execution.h:94]: I think this one is missing in the includes

Comment (corot) [reply mbf_abstract_nav/include/mbf_abstract_nav/abstract_controller_execution.h:94]: ah... shit, I forgot to update the tests :see_no_evil:

File: mbf_utility/CMakeLists.txt
Comment (corot) [reply mbf_utility/CMakeLists.txt]: yup.....  https://github.com/magazino/move_base_flex/issues/184

File: mbf_utility/src/robot_information.cpp
Comment (corot) [reply mbf_utility/src/robot_information.cpp:79]: yes, I thought about that, but... we cannot warranty that odom is properly stamped, so I didn't take the risk

but looks like we must copy odom helper (see below), so I can do a proper implementation of this check

**Comment by corot** [comment]
About the formatting....  are u ok with an extra commit formatting to clang the modified code?
some files have wrong indentation, so changes will look bigger than what they really are

Maybe I can do after having the approvals

File: mbf_utility/src/robot_information.cpp
Comment (corot) [reply mbf_utility/src/robot_information.cpp:79]: now implemented

File: mbf_abstract_nav/include/mbf_abstract_nav/abstract_controller_execution.h
Comment (corot) [reply mbf_abstract_nav/include/mbf_abstract_nav/abstract_controller_execution.h]: coopted odom helper

**Comment by corot** [comment]
Ok, I coopted odom helper. Ready for a final review, @dorezyuk, @spuetz 

And let me know what I do with the formatting;  I have the commit waiting for after your reviews

**Comment by spuetz** [comment]
Yea please change the formatting back. Formatting all the other files should be in a separate PR. I hope this is not too much work. Cheers - Seb

**Comment by corot** [comment]
> Yea please change the formatting back. Formatting all the other files should be in a separate PR. I hope this is not too much work. Cheers - Seb

No, no,,, I didn't format anything on the PR.   I was just explaining why the check fails and offering to address that

File: mbf_abstract_nav/test/abstract_controller_execution.cpp
Comment (ChristofDubs) [inline mbf_abstract_nav/test/abstract_controller_execution.cpp]: I suppose you meant `tolerances` ?

File: mbf_utility/src/robot_information.cpp
Comment (ChristofDubs) [inline mbf_utility/src/robot_information.cpp]: I think there is some room for improvement here:
- given that user code worked so far without an `odom` topic being present, this should be a warning rather than an error?
- since this is not expected to spuriously fail / if it fails, it will most likely continue to fail, use a `THROTTLE` version
- add the odom topic to the warning using `base_odom.getOdomTopic()`

On the larger picture, it seems the `RobotInformation` class in `AbstractNavigationServer` used the default parameter value of the odom topic. Would be better to expose this as a ROS parameter (default initialized to `"odom"`)? Then, the warning here could be extended to inform the user to set this parameter. (Maybe could even allow the user to suppress the warning by explicitly setting the odom topic parameter to an empty string)

File: mbf_utility/include/mbf_utility/robot_information.h
Comment (ChristofDubs) [inline mbf_utility/include/mbf_utility/robot_information.h:87]: this is unused, is that intentional?

Comment (corot) [reply mbf_utility/include/mbf_utility/robot_information.h:87]: Yes; we want to replace base_local_planner with the same functionality, and it provides this check

File: mbf_utility/src/robot_information.cpp
Comment (corot) [reply mbf_utility/src/robot_information.cpp]: good idea

**Review by ChristofDubs — APPROVED** [review]

**Pull request merged**
